server/data/data-postgres/src/main/java/org/apache/james/sieve/postgres/PostgresSieveScriptDAO.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.james.core.Username;
+import org.apache.james.sieve.postgres.model.PostgresSieveScript;
+import org.apache.james.sieverepository.api.ScriptName;
+import org.jooq.Record;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class PostgresSieveScriptDAO {
+    private final PostgresExecutor postgresExecutor;
+
+    @Inject
+    public PostgresSieveScriptDAO(@Named(DEFAULT_INJECT) PostgresExecutor postgresExecutor) {
+        this.postgresExecutor = postgresExecutor;
+    }
+
+    public Mono<Void> upsertScript(PostgresSieveScript sieveScript) {
+        return postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.insertInto(TABLE_NAME)
+            .set(USERNAME, sieveScript.getUsername())
+            .set(SCRIPT_NAME, sieveScript.getScriptName())
+            .set(SCRIPT_SIZE, sieveScript.getScriptSize())
+            .set(SCRIPT_CONTENT, sieveScript.getScriptContent())
+            .set(IS_ACTIVE, sieveScript.isActive())
+            .set(ACTIVATION_DATE_TIME, sieveScript.getActivationDateTime())
+            .onConflict(USERNAME, SCRIPT_NAME)
+            .doUpdate()
+            .set(SCRIPT_SIZE, sieveScript.getScriptSize())
+            .set(SCRIPT_CONTENT, sieveScript.getScriptContent())
+            .set(IS_ACTIVE, sieveScript.isActive())
+            .set(ACTIVATION_DATE_TIME, sieveScript.getActivationDateTime())));
+    }
+
+    public Mono<PostgresSieveScript> getScript(Username username, ScriptName scriptName) {
+        return postgresExecutor.executeRow(dslContext -> Mono.from(dslContext.selectFrom(TABLE_NAME)
+            .where(USERNAME.eq(username.asString()),
+                SCRIPT_NAME.eq(scriptName.getValue()))))
+            .map(recordToPostgresSieveScript());
+    }
+
+    public Flux<PostgresSieveScript> getScripts(Username username) {
+        return postgresExecutor.executeRows(dslContext -> Flux.from(dslContext.selectFrom(TABLE_NAME)
+                .where(USERNAME.eq(username.asString()))))
+            .map(recordToPostgresSieveScript());
+    }
+
+    public Mono<PostgresSieveScript> getActiveScript(Username username) {
+        return postgresExecutor.executeRow(dslContext -> Mono.from(dslContext.selectFrom(TABLE_NAME)
+                .where(USERNAME.eq(username.asString()),
+                    IS_ACTIVE.eq(true))))
+            .map(recordToPostgresSieveScript());
+    }
+
+    public Mono<Long> activateScript(Username username, ScriptName scriptName) {
+        return postgresExecutor.dslContext()
+            .flatMap(dslContext -> Mono.from(dslContext.update(TABLE_NAME)
+                .set(IS_ACTIVE, true)
+                .set(ACTIVATION_DATE_TIME, OffsetDateTime.now())
+                .where(USERNAME.eq(username.asString()),
+                    SCRIPT_NAME.eq(scriptName.getValue()))))
+            .cast(Long.class);
+    }
+
+    public Mono<Void> deactivateScript(Username username, ScriptName scriptName) {
```

Comment:
`Method 'deactivateScript(org.apache.james.core.Username, org.apache.james.sieverepository.api.ScriptName)' is never used `